search-api/search_api/resources/businesses.py

```diff
@@ -0,0 +1,144 @@
+# Copyright © 2020 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+from tempfile import NamedTemporaryFile
+
+from flask import Blueprint, request, jsonify, send_from_directory
+from openpyxl import Workbook
+
+from search_api.auth import jwt, authorized
+from search_api.models import (
+    Corporation,
+    CorpName,
+    Office,
+    _get_corporation_search_results,
+    _merge_corpparty_search_addr_fields,
+    _normalize_addr,
+    _format_office_typ_cd
+)
+
+
+API = Blueprint('BUSINESSES_API', __name__, url_prefix='/api/v1/businesses')
+
+
+@API.route('/search/')
+@jwt.requires_auth
+def corporation_search():
+    # TODO SY - check roles
+    # check authorization
+    # if not authorized(identifier, jwt, action=['add_comment']):
+    #     return jsonify({'message': ''}), HTTPStatus.UNAUTHORIZED
+
+    args = request.args
+    results = _get_corporation_search_results(args)
+
+    # Pagination
+    page = int(args.get("page")) if "page" in args else 1
+    results = results.paginate(int(page), 20, False)
+
+    corporations = []
+    for row in results.items:
+        result_dict = {}
+
+        result_fields = ['corp_num', 'corp_nme', 'recognition_dts', 'corp_typ_cd', 'state_typ_cd', 'postal_cd']
+
+        result_dict = {key: getattr(row, key) for key in result_fields}
+        result_dict['addr'] = _merge_corpparty_search_addr_fields(row)
+
+        corporations.append(result_dict)
+
+    return jsonify({'results': corporations})
+
+
+@API.route('/search/export/')
+@jwt.requires_auth
+def corporation_search_export():
+
+    # Query string arguments
+    args = request.args
+
+    # Fetching results
+    results = _get_corporation_search_results(args)
+
+    # Exporting to Excel
+    wb = Workbook()
```

Comment:
Hmm, you could have requested export as CSV/excel option in reports-api for this. Nvm, will move it once that feature is there in reports-api.